distro_integration.md

```diff
@@ -0,0 +1,408 @@
+   `dts/upstream/src/arm64/qcom/x1e80100-lenovo-yoga-slim7x.dts` as an example.
+2. Search for `panel` and look for a node with that name, in this case it looks
+   like:
+```dts
+&mdss_dp3 {
+    compatible = "qcom,x1e80100-dp";
+    /delete-property/ #sound-dai-cells;
+
+    status = "okay";
+
+    aux-bus {
+        panel {
+            compatible = "samsung,atna45dc02", "samsung,atna33xc20";
+            enable-gpios = <&pmc8380_3_gpios 4 GPIO_ACTIVE_HIGH>;
+            power-supply = <&vreg_edp_3p3>;
+...
+```
+3. Copy the `compatible` property strings (if there are multiple then do this
+   for both) and grep the drivers directory for them: `grep -rnwI drivers
+   -e "samsung,atna45dc02"`, you may only get a hit on one of them, this is normal.
+4. Open the Makefile in the same directory as the C file that grep points you to (we got `drivers/gpu/drm/panel/panel-samsung-atna33xc20.c:335` so open `drivers/gpu/drm/panel/Makefile`) and find the reference to the file (it won't have the `.c` suffix).
+
+```makefile
+obj-$(CONFIG_DRM_PANEL_SAMSUNG_ATNA33XC20) += panel-samsung-atna33xc20.o
+```
+
+There you go, `CONFIG_DRM_PANEL_SAMSUNG_ATNA33XC20` is what we need to enable for the panel driver in the Yoga Slim 7x.
+
+#### The pesky last few
+
+There's always a few things missing that are a total pain to track down, you can
+check if there's anything in `/sys/kernel/debug/devices_deferred`, though note
+that missing firmware (a likelehood if you aren't on a ThinkPad) will cause some
+drivers not to probe, we'll come back to these later.
+
+Don't be afraid to `grep` for compatible strings in the `drivers/` directory of
+the kernel, you can also usually work backwords from a C file, through the
+`Makefile` in the same directory and to the associated kernel config option
+which you need.
+
+And of course, head over to the IRC where folks can probably help you translate
+your error message or missing functionality into the necessary config options.
+
+## Part 2 - Firmware
+
+Many of the benefits of ARM systems come from their heavy use of co-processors
+for tasks like audio processing, compute, video decoding, etc (notably things
+that the x86 world has been trying to catch up with in the last few years).
+
+But these co-processors all need to run firmware, and that firmware is usually
+tied to your specific device. Some of these are packaged in
+[linux-firmware](https://web.git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/)
+but for most devices there is no clear license and such packaging isn't
+possible. Instead the firmware must be pulled from the Windows partition (and
+dual booting is the only way to get firmware updates). The only "correct" way to
+handle this is to have your users dual-boot, and to periodically boot back into
+Windows for updates, before then copying over new firmware versions.
+
+For X1E laptops, the
+[woa-firmware-yoinker](https://github.com/CadmiumLinux/woa-firmware-yoinker/blob/main/yoink-firmware)
```

Comment:
there is also some project by the ubuntu maintainers (i could find a link if you care about adding it)

Reply:
Would be good to add for sure!

Reply:
https://salsa.debian.org/debian/qcom-firmware-extract

Reply:
Thanks! Added that too.